api.js

```diff
@@ -296,14 +195,141 @@ const api = {
     }
   },
 
+  // @summary loans glimmer from one user to another
+  // @param userId - calling user
+  // @param amount - the amount the user chose to loan
+  // @param loanTo - the user to loan amount to
+  // @param bot - this bot, duh
+  // @channelId - id of the channel to write to
+  loan(userId, amount, loanTo, bot, channelId) {
+    try {
+      let message = ``;
+
+      if (isNaN(amount)) {
+        bot.sendMessage({
+          to: channelId,
+          message: `<@${userId}> ${amount} isn't a number, dumbass.`
+        });
+
+        return;
+      } else if (amount < 0) {
+        bot.sendMessage({
+          to: channelId,
+          message: `<@${userId}> Nice try dipshit`
+        });
+
+        return;
+      }
+
+      const user = this.database.ref(`users/${userId}`);
+      const userLoanTo = this.database.ref(`users/${loanTo}`);
+      user.once('value', snapshot => {
+        if (snapshot.val()) {
+          if (snapshot.val().glimmer < amount) {
+            bot.sendMessage({
+              to: channelId,
+              message: `<@${userId}> you don't have that much glimmer to loan dickface.`
+            });
+            return;
+          }
+          bot.sendMessage({
+            to: channelId,
+            message: `<@${userId}> Good financial decision`
+          });
+          let glimmer = Number(snapshot.val().glimmer) - Number(amount);
+          snapshot.ref.update({ glimmer });
+        }
+      });
+
+      userLoanTo.once('value', snapshot => {
+        if (snapshot.val()) {
+          bot.sendMessage({
+            to: channelId,
+            message: `<@${loanTo}> You just got ${amount} glimmer! Good luck...`
+          });
+
+          if (!(isNaN(snapshot.val().oweTo.userId))) {
+            let owed = Number(snapshot.val().oweTo.userId) + Number(amount);
+            let oweTo = {userId: Number(owed)};
```

Comment:
Think this still has that bug we talked about last night, you want oweTo[userId] = Number(owed) instead of { userId: Number(owed) }. Also removed the console.log please.